### ISSUE TITLE
contextnet model not saved after training the model

### ISSUE THREAD

**Issue by vivekyadav21** [TensorSpeech/TensorFlowASR#201]
hi
i trained the contextnet model for my custom datasets with mfcc featurizer and changing the required directories in config file and leaving the rest as it is.
But after traning for 20 epochs no model saved 
@usimarit please help me

**Comment by usimarit** [comment]
@vivekyadav21 To make sure you save checkpoint, please create the folder first and use `save_best_only: False`.

**Comment by vivekyadav21** [comment]
thanku @usimarit it is saved now

**Comment by vivekyadav21** [comment]
@usimarit hi i have couple of questions to ask 
1.why this below error occur: **padding must be less than dimension size,256 is not less than 25**
  although this issue solved by removing less duration audios but  i don't know exactly where the problem is
![Uploading Screenshot from 2021-05-28 14-06-40.png…]()

2. can we use ctc word beam search with contextnet and what is maximum beam width we can use
@usimarit THANKU SO MUCH FOR SUCH AN AMAZING REPO

**Comment by vivekyadav21** [comment]
@usimarit please take a look here.

**Comment by usimarit** [comment]
@vivekyadav21 
1. How long are the less duration audios that you removed?
2. No, ctc beam search is for ctc models only, you can use any beam width, as long as your machine can handle it 😄

**Comment by vivekyadav21** [comment]
@usimarit thanku for your reply 
1.since my dataset has very less rows with duration 0.5 sec or less ,so i removed all those audios which are less than 0.5 sec.
Have another question do you have any plan for implementing word based beam search in transducer ?
Is there any hack or trick  to speed up this beam search decoding  time ,it taking lot of time with just a beam width of 5?

**Comment by usimarit** [comment]
@vivekyadav21 I'm busy lately so I don't have any plans for word based beam search. I tried to optimize the current beam search as much as possible but I think to speed up more, we will have to apply the new method (n-step constrained beam search).

**Comment by vivekyadav21** [comment]
@thanku for suggestion and reply

**Comment by usimarit** [comment]
I’ll close the issue here. Feel free to reopen if you have any more questions.
